File: LeetCode/448. Find All Numbers Disappeared in an Array.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size() ;
        vector<int> ans ;
        map<int,bool> m ;
        for(int i = 0 ; i < n ; i++)
        {
            m[nums[i]] = true ;
        }
        for(int i = 1 ; i <= n ; i++)
        {
            if(m.find(i) == m.end())
            {
                ans.push_back(i) ;
            }
        }
        return ans ;
    }
};

File: LeetCode/448. Find All Numbers Disappeared in an Array.java
/***  KNOWLEDGE IS POWER  ***/

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int arr[]=new int[nums.length+1];
        List<Integer> list=new ArrayList<Integer>();
        for(int i=0;i<nums.length;i++){
            int temp=Math.abs(nums[i]);
            if(nums[temp-1]>0){
                nums[temp-1]=-nums[temp-1];
            }
                
        }
        for(int i=0;i<nums.length;i++){
            if(nums[i]>=0)
                list.add(i+1);
        }
        return list;
    }
}
